src/api/erc20/Erc20ApiMock.ts

```diff
@@ -82,16 +80,47 @@ export class Erc20ApiMock implements Erc20Api {
     this._initBalances(fromAddress, tokenAddress)
     this._initBalances(toAddress, tokenAddress)
 
+    // Since we don't have a msg.sender, we mimic it by using the optional field
+    const { senderAddress: spenderAddress } = txOptionalParams
+    assert(spenderAddress, 'TxOptionalParams.senderAddress is required for this mock work')
+    this._initAllowances(fromAddress, tokenAddress, spenderAddress)
+
     const balance = this._balances[fromAddress][tokenAddress]
     assert(balance.gte(amount), "The user doesn't have enough balance")
 
+    assert(
+      this._checkAndUpdateAllowance(spenderAddress, fromAddress, tokenAddress, amount),
```

Comment:
I think should never update anything. I understand the method should be called `_hasAllowance`, then after the assert, we update the allowances.

Reply:
Ok, will split it.